WalletWasabi.Fluent/ViewModels/CoinControl/SelectCoinsDialogViewModel.cs

```diff
@@ -0,0 +1,168 @@
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using Avalonia.Controls;
+using Avalonia.Controls.Models.TreeDataGrid;
+using Avalonia.Controls.Templates;
+using NBitcoin;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Fluent.Helpers;
+using WalletWasabi.Fluent.Models;
+using WalletWasabi.Fluent.TreeDataGrid;
+using WalletWasabi.Fluent.ViewModels.CoinControl.Core;
+using WalletWasabi.Fluent.ViewModels.Dialogs.Base;
+using WalletWasabi.Fluent.ViewModels.Wallets;
+using WalletWasabi.Fluent.Views.CoinControl.Core.Cells;
+using WalletWasabi.Fluent.Views.CoinControl.Core.Headers;
+
+namespace WalletWasabi.Fluent.ViewModels.CoinControl;
+
+[NavigationMetaData(
+	Title = "Coin Selection",
+	Caption = "",
+	IconName = "wallet_action_send",
+	NavBarPosition = NavBarPosition.None,
+	Searchable = false,
+	NavigationTarget = NavigationTarget.DialogScreen)]
+public partial class SelectCoinsDialogViewModel : DialogViewModelBase<IEnumerable<SmartCoin>>
+{
+	public SelectCoinsDialogViewModel(WalletViewModel walletViewModel)
+	{
+		var pockets = walletViewModel.Wallet.GetPockets();
+		var items = CreateItems(pockets);
+
+		var pocketColumn = PocketColumn();
+		Source = new HierarchicalTreeDataGridSource<CoinControlItemViewModelBase>(items)
+		{
+			Columns =
+			{
+				ChildrenColumn(),
+				IndicatorsColumn(),
+				AmountColumn(),
+				PrivacyScore(),
+				pocketColumn
+			}
+		};
+		Source.SortBy(pocketColumn, ListSortDirection.Descending);
+		Source.RowSelection!.SingleSelect = true;
+		Source.RowSelection!.SingleSelect = true;
```

Comment:
Duplication. It was a mistake in my suggestions previously.

```suggestion
		Source.RowSelection!.SingleSelect = true;
```